azure/services/natgateways/natgateways.go

```diff
@@ -118,12 +122,27 @@ func (s *Service) getExisting(ctx context.Context, spec azure.NatGatewaySpec) (*
 	if err != nil {
 		return nil, err
 	}
+	// We must have a non-nil, non-"empty" PublicIPAddresses
+	if !(existingNatGateway.PublicIPAddresses != nil && len(*existingNatGateway.PublicIPAddresses) > 0) {
```

Comment:
@CecileRobertMichon no longer requiring exactly 1 item in the `existingNatGateway.PublicIPAddresses` slice